File: Data Structures & Algorithms/two-integer-sum-ii/submission-8.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        """
        numbers = [1,2,3,4], target = 3
                   l     r
        """
        l, r = 0, len(numbers) - 1
        
        while l < r:
            if (numbers[l] + numbers[r]) == target:
                return [l + 1, r + 1]
            elif (numbers[l] + numbers[r]) > target:
                r -= 1
            else:
                l += 1

File: Data Structures & Algorithms/binary-search/submission-8.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size();
        while (l < r) {
            int middle = l + (r - l)/2;
            if (nums[middle] > target) {
                r = middle;
            }
            else {
                l = middle + 1;
            }
        }
        return (l > 0 && nums[l - 1] == target) ? l - 1: -1;
    }
};
